File: src/csharp_problems/medium/p2216-delete-the-middle-node-of-a-linked-list/delete-the-middle-node-of-a-linked-list.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode DeleteMiddle(ListNode head) {
        if (head.next == null) return null;

        ListNode node = head;
        int count = 0;
        while (node != null) {
            node = node.next;
            count++;
        }
        int index = count / 2;
        node = head;
        for (int i = 1; i < index; i++) node = node.next;
        Console.WriteLine("c:" + node.val);
        node.next = node.next.next;

        return head;
    }
}

File: 2216-delete-the-middle-node-of-a-linked-list/delete-the-middle-node-of-a-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* deleteMiddle(struct ListNode* head) {
    if (head->next == NULL) return NULL;

    struct ListNode* node = head->next;
    struct ListNode* haveList = head;
    int i = 1;
    while (node->next && node->next->next) {
        haveList = haveList->next;
        node = node->next->next;
    }

    haveList->next = haveList->next->next;
    return head;
}